bentoml/_internal/service/inference_api.py

```diff
@@ -39,7 +45,77 @@ def __init__(
         InferenceAPI._validate_route(route)
 
         self.name = name
+        self.needs_ctx = False
+        self.ctx_param = None
         self.func = user_defined_callback
+        input_type = input_descriptor.input_type()
+        self.multi_input = isinstance(input_type, dict)
+        sig = inspect.signature(user_defined_callback)
+        if isinstance(input_type, dict):
```

Comment:
same here

Reply:
or just use `if self.multi_input` here?

Reply:
Pyright complains if I use `self.multi_input`, so I gave up and used this instead.